llvm/lib/Transforms/Instrumentation/AddressSanitizer.cpp

```diff
@@ -1750,12 +1895,21 @@ void AddressSanitizer::instrumentAddress(Instruction *OrigIns,
 
   Value *AddrLong = IRB.CreatePointerCast(Addr, IntptrTy);
   if (UseCalls) {
-    if (Exp == 0)
-      IRB.CreateCall(AsanMemoryAccessCallback[IsWrite][0][AccessSizeIndex],
-                     AddrLong);
-    else
+    if (Exp == 0) {
+      if (TargetTriple.isSPIR()) {
+        SmallVector<Value *> Args;
```

Comment:
If you can estimate how many args there are. It's better to pass a number N to SmallVector.

Reply:
done

Reply:
I can't give it a fix number of SmallVector, because this will conflict with the argument type of AppendDebugInfoToArgs

Reply:
The usual pattern for this would be to have `AppendDebugInfoToArgs` accept `SmallVectorImpl<Value *>&` instead of `SmallVector<Value *>&`. That way `AppendDebugInfoToArgs` will accept any `SmallVector<Value *, N>`.

Reply:
See note in `SmallVector` section of [LLVM Programmer's Manual](https://llvm.org/docs/ProgrammersManual.html#llvm-adt-smallvector-h). `AppendDebugInfoToArgs` should accept `SmallVectorImpl<T> &` and then you will be able to specify second template argument `N` here

Reply:
@kbenzie @AlexeySachkov Thank you very much :)